pkg/lib/configreader/string.go

```diff
@@ -62,6 +62,7 @@ type StringValidation struct {
 	CastScalar                           bool
 	AllowCortexResources                 bool
 	RequireCortexResources               bool
+	DockerImage                          bool
```

Comment:
what is this for? user container image? In any case, I think it needs a better naming scheme, since docker is a company

Reply:
It checks whether it's a valid docker image registry path. We can do this in a separate PR as this is a literal copy-paste from the other configreader validations.